packages/backend/src/graphql/schema.graphql

```diff
@@ -900,6 +903,44 @@ type WebhookEventsEdge {
   cursor: String!
 }
 
+enum FeeType {
+  "Sender pays the fees"
+  SENDING
+  "Receiver pays the fees"
+  RECEIVING
+}
+
+type Fee {
+  "Fee id"
+  id: ID!
+  "Asset id associated with the fee"
+  assetId: ID!
+  "Type of fee (sending or receiving)"
+  type: FeeType!
+  "Fixed fee"
+  fixed: UInt64
+  "Percentage fee"
+  percentage: Float
+  "Date-time of creation"
+  createdAt: String!
+}
+
+input UpdateFeeInput {
+  "Asset id to add the fee to"
+  assetId: ID!
+  "Type of fee (sending or receiving)"
+  type: FeeType!
+  "Fee values"
+  fee: FeeDetails!
+}
+
+input FeeDetails {
+  "A flat fee"
+  fixed: UInt64
+  "Percentage fee"
+  percentage: Float
```

Comment:
What do you think about making these required in the API? 
I kind of like the idea given the fact that there is no guessing whether an input like this:
```
{
    "input": {
        "assetId": "{{assetId}}",
        "type": "SENDING",
        "fee": {
          "percentage": 0.01
        }
    }
}
```
would update or not update a previously set `fixed` fee

Reply:
I think that's probably better. Making both required and using `0` to represent no fixed/percentage fee instead. I find that more intuitive for the case you highlight, and it will get rid of some extra validation logic to ensure one is supplied if both are optional.

Reply:
both are required now